### DISCUSSION TITLE
How to print the detailed information of equations and variables during the FMU generation process?

### DISCUSSION THREAD

**Discussion by Zbm1996** [OpenModelica/OpenModelica#10242] [Q&A]
Currently, I used omc executable file to generate CS2.0 FMU based on one huge model. Within the mos file, I used checkModel to print the number of equations and variables.The strange thing was that I found the number of variables was bigger than the equations but the fmu was generated successfully. 
![不匹配](https://user-images.githubusercontent.com/25106682/220341464-4ad8d549-a20e-4ac5-b34b-79bf8c74045c.PNG)
However, this fmu still failed in our own simulator. It was trapped in dead loop while calling functionODE or funtionDAE like this.
![死循环](https://user-images.githubusercontent.com/25106682/220342390-b0c5001d-aabc-402e-a8d3-a0fe34ed5f42.PNG)
The related function is "internalEventIteration" of function "fmi2DoStep" in file "fmu2_model_interface.c.inc".

Are there any command in mos to print the detailed information of equations and variables not just the number? Or do you guys ever met such kind of problem and find it was not because of the no match of equations and variables?

It seemed that the fmu always failed in initialization process while calling some non linear solver but I'm not sure.

This problem really annoyed me for adays and I'm still not sured whether I was wrong in model building or are there any issue for OpenModelica to solve.

**Comment by AnHeuermann** [comment]
You can provide debug flag [`-d=dumpSimCode`](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/omchelptext.html#omcflag-debug-dumpsimcode)  to let the OpenModelica compiler spit out all equations that are used to generate the C code. 

```
loadModel(Modelica, {"4.0.0"}); getErrorString();
setCommandLineOptions("-d=dumpSimCode"); getErrorString();
buildModelFMU(Modelica.Mechanics.MultiBody.Examples.Elementary.Pendulum); getErrorString();
```

Search for
```
*********************
* SimCode Equations *
*********************


allEquations:
========================================
557: $DER.damper.phi_rel=rev.w [Real]
558: damper.w_rel=rev.w [Real]
```

There is no easy way to get the original Modelica equation from the simCode equations.

There is also the Transformation Debugger of OMEdit, see https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/debugger.html#use-the-transformation-debugger-for-browsing.
You can find the equation index, the simCode equation as well as the Modelica equation where it is comming from. Problem is, that you are using FMU export. It could be, that the equation numbers are different compared to the "normal" simulation done by OMEdit.
